migrations/20241105164503_2.sql

```diff
@@ -0,0 +1,40 @@
+-- Force anything running this migration to use the right search path.
+set local search_path to underway;
+
+-- Remove attempt-specific columns from `task`
+alter table underway.task
+drop column if exists retry_count,
+drop column if exists initial_interval_ms,
+drop column if exists max_interval_ms,
+drop column if exists backoff_coefficient,
+drop column if exists error_message,
+drop column if exists started_at,
+drop column if exists succeeded_at,
+drop column if exists last_failed_at;
+
+alter table underway.task
+add column if not exists retry_policy jsonb not null default '{"max_attempts": 5, "initial_interval_ms": 1000, "max_interval_ms": 60000, "backoff_coefficient": 2.0}';
```

Comment:
I am not big fan of json, knowning the shape of data seems helps to reason about things better.
I do understand that this is more work and maybe doesnt worth it, but wdyt about having custom type here?

Reply:
Aha, I didn't realize this was possible. Yes, I'll look at implement it that way. Thank you.

Reply:
but again, i am not sure its worth tbh.
i now think probably it is not...

Reply:
I think it's not too difficult, something like this:

```sql
-- Define retry policies as their own type
create type underway.task_retry_policy as (
    max_attempts         int,
    initial_interval_ms  int,
    max_interval_ms      int,
    backoff_coefficient  float
);

alter table underway.task
    add column if not exists retry_policy underway.task_retry_policy not null 
    default row(5, 1000, 60000, 2.0)::underway.task_retry_policy;
```

And we can derive the type with SQLx:

```rust
#[derive(Debug, Clone, Copy, PartialEq, sqlx::Type)]
#[sqlx(type_name = "task_retry_policy")]
pub struct RetryPolicy {
    pub(crate) max_attempts: i32,
    pub(crate) initial_interval_ms: i32,
    pub(crate) max_interval_ms: i32,
    pub(crate) backoff_coefficient: f32,
}
```

Reply:
This is added. It was pretty straightforward to do. I'm not sure if there are downsides I might be overlooking here but it seems nice to create the custom type as you suggested.